lxd/network/driver_ovn.go

```diff
@@ -5887,3 +5907,76 @@ func (n *ovn) checkAddressNotInUse(netip *net.IPNet) (bool, error) {
 
 	return true, nil
 }
+
+// checkAddressNotInOVNRange checks that a given IP address does not overlap
+// with OVN ranges set on the uplink network. However, it allows an IP address
+// that is the same as volatile network address of the current network.
+// Returns an error if the check could not be performed or the IP address
+// overlaps with OVN ranges.
+func (n *ovn) checkAddressNotInOVNRange(addr net.IP) error {
+	if addr == nil {
+		return fmt.Errorf("Invalid listen address")
+	}
+
+	if n.config["network"] == "" {
+		return fmt.Errorf(`OVN network %q is missing "network" config option`, n.name)
+	}
+
+	addrIsIP4 := addr.To4() != nil
+
+	ovnRangesKey := "ipv4.ovn.ranges"
+	volatileNetworkAddrKey := "volatile.network.ipv4.address"
+
+	if !addrIsIP4 {
+		ovnRangesKey = "ipv6.ovn.ranges"
+		volatileNetworkAddrKey = "volatile.network.ipv6.address"
+	}
+
+	// It is acceptable to set up a network forward or a load balancer using the same
+	// volatile network address as the associated OVN network.
+	if n.config[volatileNetworkAddrKey] != "" {
+		volatileAddr := net.ParseIP(n.config[volatileNetworkAddrKey])
+
+		if addr.Equal(volatileAddr) {
+			return nil
+		}
+	}
+
+	var uplink *api.Network
+	var err error
+
+	uplinkName := n.config["network"]
```

Comment:
Again, just a nit, but i would suggest defining this var above the check for `if n.config["network"] == "" {` and then changing to `if uplinkName == "" {` so you're testing the same var you use further down to avoid the chance of the two diverging later on.